packages/extension/src/ui/features/accountTokens/tokens.state.ts

```diff
@@ -162,16 +162,16 @@ export const useFungibleTokens = (
     return selectedAccount?.networkId ?? ""
   }, [selectedAccount?.networkId])
 
+  const sortedTokenIds = allUserTokens.map((t) => t.id).sort()
   const cachedTokens = useTokensInNetwork(networkId)
   const {
     data: fungibleTokens,
     isValidating,
     error
-  } = useSWRImmutable(
+  } = useSWR(
     selectedAccount && [
       getAccountIdentifier(selectedAccount),
-      allUserTokens.map((t) => { return { id: t.id, balance: t.balance.toString() } }),
-      cachedTokens,
+      sortedTokenIds,
       "accountFungibleTokens",
```

Comment:
The keys are still unique? So it seems we could use immutable SWR?

Reply:
The balance of these tokens could change, so if we use immutable SWR I guess it could always return the same balance for fungible tokens.

Reply:
Sorry, I thought this function was to get token metadata. `useSWR` is right